### ISSUE TITLE
Craft CMS 4.12.4.1 Getting unknown property: craft\awss3\Fs::rootPath

### ISSUE THREAD

**Issue by phillipsandoveracademy** [DeuxHuitHuit/craft-cloudflare-stream#15]
I am getting the error `Getting unknown property: craft\awss3\Fs::rootPath` on my local DDEV running Craft CMS 4.12.4.1, Amazon S3 2.2.1, and Cloudflare Stream 1.6.0. Below is the stack trace.

```
yii\base\UnknownPropertyException: Getting unknown property: craft\awss3\Fs::rootPath in /var/www/html/vendor/yiisoft/yii2/base/Component.php:154
Stack trace:
#0 /var/www/html/vendor/deuxhuithuit/craft-cloudflare-stream/src/Folder.php(13): yii\base\Component->__get('rootPath')
#1 /var/www/html/vendor/deuxhuithuit/craft-cloudflare-stream/src/controllers/UploadController.php(58): deuxhuithuit\cfstream\Folder::getAssetFolderPath(Object(craft\elements\Asset))
#2 [internal function]: deuxhuithuit\cfstream\controllers\UploadController->actionUpload()
#3 /var/www/html/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#4 /var/www/html/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams(Array)
#5 /var/www/html/vendor/yiisoft/yii2/base/Module.php(552): yii\base\Controller->runAction('upload', Array)
#6 /var/www/html/vendor/craftcms/cms/src/web/Application.php(341): yii\base\Module->runAction('cloudflare-stre...', Array)
#7 /var/www/html/vendor/craftcms/cms/src/web/Application.php(640): craft\web\Application->runAction('cloudflare-stre...', Array)
#8 /var/www/html/vendor/craftcms/cms/src/web/Application.php(303): craft\web\Application->_processActionRequest(Object(craft\web\Request))
#9 /var/www/html/vendor/yiisoft/yii2/base/Application.php(384): craft\web\Application->handleRequest(Object(craft\web\Request))
#10 /var/www/html/public/index.php(26): yii\base\Application->run()
#11 {main}
```

This error occurs when I click the Upload to Cloudflare button. I removed auto-upload as this issue was blocking the original upload of the video into the Craft CMS filesystem.

*UPDATE*
**p1—** This issue happens on our staging server as well.
**p2—** I was successfully able to upload a video on our staging server from a local folder, instead of AWS.

I am not sure if I have my setup (local or stage) incorrect. Is this a bug? or a setup issue on my part? I do realize I am running Craft CMS 4, but I am not quite ready to jump to Craft CMS 5 yet. Also, I believe Cloudflare Stream doesn't allow uploads from a private server (DDEV, etc.). Much appreciated!

**Comment by nitriques** [comment]
Wow this bug is a plague. We have a fix for it in v2, I'll backport it to v1.

> I believe Cloudflare Stream doesn't allow uploads from a private server (DDEV, etc.)

If you use the "upload usign form data" option, it should work ;)

**Comment by nitriques** [comment]
Re: #8

**Comment by nitriques** [comment]
Fixed in 9a8ca6fc33 released as `1.6.1` .

Thanks !
